frontend/src/tests/utils/useBackend.test.js

```diff
@@ -0,0 +1,52 @@
+import { QueryClient, QueryClientProvider } from "react-query";
+import { renderHook } from '@testing-library/react-hooks'
+import mockConsole from "jest-mock-console";
+
+import axios from "axios";
+import AxiosMockAdapter from "axios-mock-adapter";
+import useBackend from "main/utils/useBackend";
+
+
+jest.mock('react-router-dom');
+
+describe("utils/useBackend tests", () => {
+
+    test("test useBackend handles 404 error correctly", async () => {
+  
+        // See: https://react-query.tanstack.com/guides/testing#turn-off-retries
+        const queryClient = new QueryClient({
+            defaultOptions: {
+                queries: {
+                    // ✅ turns retries off
+                    retry: false,
+                },
+            },
+        })
+        const wrapper = ({ children }) => (
+            <QueryClientProvider client={queryClient}>
+                {children}
+            </QueryClientProvider>
+        );
+
+        var axiosMock = new AxiosMockAdapter(axios);
```

Comment:
Maybe we can also make "axiosMock" constant as well

Reply:
good catch. I did this as a global change in the code base